shared_task_code/ecs_task_queue_consumer.py

```diff
@@ -0,0 +1,91 @@
+import os
+import aioboto3
+import boto3
+from utils.lambda_decorators import ssm_parameters, async_handler
+from utils.json_serialisation import dumps
+from json import loads
+from datetime import datetime
+import subprocess
+from shared_task_code.task_queue_consumer import TaskQueueConsumer
+
+
+class ECSTaskQueueConsumer(TaskQueueConsumer):
+
+    def __init__(self, event):
+        self.ecs_client = boto3.client("ecs", region_name=self.region)
+
+        self.PRIVATE_SUBNETS = f"{self.ssm_prefix}/vpc/using_private_subnets"
+        self.SUBNETS = f"{self.ssm_prefix}/vpc/subnets/instance"
+        self.CLUSTER = f"{self.ssm_prefix}/ecs/cluster"
+        self.SECURITY_GROUP = f"{self.ssm_prefix}/tasks/{self.task_name}/security_group/id"
+        self.IMAGE_ID = f"{self.ssm_prefix}/tasks/{self.task_name}/image/id"
+        self.event = event
+
+    def submit_ecs_task(self, task_params, message_id):
+        ssm_params = self.event["ssm_params"]
+        private_subnet = "true" == ssm_params[self.PRIVATE_SUBNETS]
+        network_configuration = {
+            "awsvpcConfiguration": {
+                "subnets": ssm_params[self.SUBNETS].split(","),
+                "securityGroups": [ssm_params[self.SECURITY_GROUP]],
+                "assignPublicIp": "DISABLED" if private_subnet else "ENABLED"
+            }
+        }
+        env_obj = [
+            {
+                "name": "TASK_INPUT",

```

Comment:
I thought we agreed that we would allow the user to override the mapping from the input event to a set of environment variables.  Here we are relying on another piece of code to tell us what task_params to use, and is sending them all through a single environment variable.